### PR TITLE
HWP-307 Frontend Refresh Token storage

### PR DESCRIPTION
**Pull request by ZachBourque**
# Description

This PR includes the following proposed change(s):

- Added logout to authDuck.js
- Modified sideMenu.js to call logout action
- Added credentials: "include" to login and logout redux actions

## Notes

This PR stores the refresh token as an httpOnly cookie, currently said cookie is not being used to create a new access token. That will be done when the custom axios instance is implemented.

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [x] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/client/src/redux/ducks/authDuck.js [hunk 1]
```diff
@@ -88,6 +87,29 @@ export const register = (name, email, password) => async (dispatch) => {
   }
 };
 
+export const logout = () => async (dispatch) => {
```
Comment (BradyMitch) [inline app/client/src/redux/ducks/authDuck.js:90]: Logout needs to clear the access token from redux memory

**Comment by BradyMitch** [comment]
How were you going to implement the refresh? Was it that if a request received a 401 'Not Authorized' response, that you would try to refresh the token and try again?

File: app/client/src/redux/ducks/authDuck.js (hunk 1 above)
Comment (ZachBourque) [reply app/client/src/redux/ducks/authDuck.js:90]: The logout action dispatches LOGOUT which clears the auth state

**Comment by ZachBourque** [comment]
> 

Was thinking that as it could be faster for all calls. But was also considering intercepting before each call and decoding the access token to see if its expired, and if so then getting a new one

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**